src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -441,6 +488,24 @@ def sql_query(self) -> str:
         self._sql_query = f"{analyzer_utils.SELECT}{self.projection_in_str}{analyzer_utils.FROM}{from_clause}{where_clause}{order_by_clause}{limit_clause}{offset_clause}"
         return self._sql_query
 
+    @property
+    def query_params(self) -> Optional[Sequence[Any]]:
+        if self._query_params is not None:
+            return self._query_params
+
+        if not self.has_clause and not self.projection:
+            self._query_params = self.from_.query_params
+            return self._query_params
+
+        query_params = []
+        if self.params:
+            query_params.extend(self.params)
```

Comment:
can we add a test that triggers this code path

Reply:
Good catch. It does not seem like this path can be hit because we don't initiate `SelectStatement` with `params`. I'm just going to remove `self.params` altogether.